oak_functions/client/rust/src/attestation.rs

```diff
@@ -82,34 +81,42 @@ impl Client {
 /// gRPC Attestation Service client implementation.
 pub struct AttestationClient {
     client: Client,
-    encryptor: AeadEncryptor,
+    attested_state_machine: AttestationStateMachine<AttestedState>,
```

Comment:
Why does this struct keep an instance of the `AttestationStateMachine`? `AttestedState` is the final state of the state machine, so the `encrypt` and `decrypt` methods don't need to be defined on the state machine, they could be directly defined on `AttestedState`; and `AttestationClient` could simply keep an instance of `AttestedState`. 

Please also see my other comments on `AttestationStateMachine`.

Reply:
`AttestedState` was changes to `AttestedSession`, and now it's not an attestation state.